### DISCUSSION TITLE
Thumbnails view improvements: faster, fix lost cursor

### DISCUSSION THREAD

**Discussion by cryham** [doublecmd/doublecmd#1094] [Suggestions]
### Description

I've been using Thumbnail view for years now as my main way of browsing images.
I simply find it much more convenient than starting other program with own GUI, own shorcuts, own or no bookmarks etc.

Current Thumbnail view is quite slow compared to other dedicated image browser softwares.
I'd suggest generating thumbnails in DC on many threads (same count as CPUs on PC), thus making it like 8 times faster in my case.

Another bug that I constantly have is that Thumbnail generation cursor is lost and doesn't update quite often.
E.g.: I enter a folder, it starts at 0, I press PgDn slowly few times, it catches to new location, but then I press PgDn and hold for few seconds, and it is still in previous location and I don't see thumbs where I am now.
It should always immediately start generating thumbnails in current location (on many threads). It should not matter if I'm holding PgDn, it should then be changed 30 times per second without loosing itself. This is working in other programs and I can immediately see few pictures thumbs to have some orientation at speed.
DC does loose thumbs generation often and sometimes moving files (F6) helps it start from current location, other times you'd need to enter folder again. It's usually quicker to press Enter and see an image than to fix it or wait for thumbs.

Lastly, Thumbnail generation should always continue in background. Currently it stops when all in view are generated. It should continue generating next (or previous if I was going up or PgUp) so that likely later when I move there, thumbs will be already generated and show up quickly.

Similar to 1 issue from 2016: https://doublecmd.sourceforge.io/mantisbt/view.php?id=1438

**Comment by JoePalffy** [comment]
> Thumbnail generation should always continue in background
I'm not so sure about that one, maybe your collections are not very big or you only view thumbnails with images, but generating thumbnails for everything is a waste of space and processing power, also this is not just down to the CPU, imagine how slow a 8-16TB SMR HDD would be at this? so this should be an option not a requirement.
